spec.md

```diff
@@ -841,13 +841,113 @@ For success responses, the following fields are defined:
+| server_url | string | A URI pointing to a remote server where API extensions will run. This URI will be used as the basepath for the endpoint(s) described by the `discovery_url` OpenAPI document. If no `server_url` is present, the Platform MUST assume the extension API endpoint(s) are to be invoked using the Service Broker host and port. If present, MUST be a valid URI. If `server_url` is a path, the Platform can assume it is to be called relative to the basepath of the Service Broker. |
+| credentials | object | A Service Broker MAY return authentication details for running any of the extension API calls, especially for those running on remote servers. If not present, the same authentication mechanism used for the normal Open Service Broker APIs MUST work for the new endpoint(s). If the Service Broker wants to use alternate methods of authentication, (e.g. on remote servers) it MUST provide details to that mechanism in the OpenAPI document via one or more [Security Scheme Objects](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#securitySchemeObject). Next, each extension endpoint in the OpenAPI document MUST have one or more [Security Requirement Objects](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#securityRequirementObject) defined that matches a [Security Scheme Object](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#securitySchemeObject). Lastly, the parameters needed by each [Security Scheme Object](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#securitySchemeObject), (e.g. username and password for basic authentication), MUST be provided as part of the `extension_api` object within the `credentials` object. If credentials are present in an `extension_api` object, the Platform will need to verify the authentication method from the OpenAPI document. |
+| alt_instance_id | string | Refers to a parameter in the `discovery_url` OpenAPI document that maps directly to the Service Broker API `instance_id` parameter. If the extension API endpoint(s) use a different value to represent a Service Broker instance, then `alt_instance_id` MUST be present. If not present, the Platform can assume `instance_id` means Service Broker Instance in the OpenAPI document. |
+| operation_display_names | object | A Service Broker MAY match operations in the `discovery_url` OpenAPI document with user consumable display names. If present, each key in this object MUST match an [operationID](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#operationObject) from the `discovery_url` OpenAPI document, with each value representing a human readable string. Platforms can use this object to show end users operations that are available on their Service Instance prior to calling the `discovery_url` OpenAPI document. |
```

Comment:
s/calling/retrieving/

Reply:
How does this play with the idea of more than one broker supporting the same set of APIs? In particular, what if each provided their own set of display names and they're all different? As long as the platform never tries to do any kind of bulk action I guess that's ok - but even so, seeing a different phrase for the same semantics might give a funky UX.

While I understand the desire for this, it sure would be nice if we could say something like "use the last phrase in the operation's URL/path".

Reply:
I think there are multiple implementation paths for platforms to accept the extensions api:

1. Handle everything dynamically; use 'operation_display_names' to show operations before generating UI forms from the openapi doc.
2. Have a platform specific definition for how operations and openapi documents are written. The platform can create pre-generated UIs based on known operation names and parameters.
3. Utilizes adheres_to to pre-generate UIs on a case by case basis according to some standard(s).
4. All of the above.

IMO, it's pick your poison and 'operation_display_names' gives some UI advantage for #1 in that you don't need to load up the OpenAPI doc prior to generating the form for the specific operation.